src/nvim/message.c

```diff
@@ -3064,14 +3072,19 @@ int msg_end(void)
 
 void msg_ext_ui_flush(void)
 {
-  if (!ui_has(kUIMessages)) {
+  if (!msg_enable_ext()) {
     return;
   }
 
   msg_ext_emit_chunk();
   if (msg_ext_chunks.size > 0) {
-    ui_call_msg_show(cstr_to_string(msg_ext_kind),
-                     msg_ext_chunks, msg_ext_overwrite);
+    if (ui_has(kUIMessages)) {
+      ui_call_msg_show(cstr_to_string(msg_ext_kind),
+                       msg_ext_chunks, msg_ext_overwrite);
+    } else {
+      msg_call_msgfunc("msg_show", &msg_ext_chunks);
```

Comment:
`ui_call_msg_show()` calls `ui_event()`, which frees the allocated chunk here:
<https://github.com/neovim/neovim/blob/120a88163078e61b272e9629138642bc5df80e4a/src/nvim/ui.c#L180-L182>
So the allocated chunk should also need freeing in or after `msg_call_msgfunc()`.

Reply:
OK.